src/Infrastructure/Services/OfficeService/OfficeService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Application.DTOs.OfficeDTO;
+using AutoMapper;
+using Domain.Entities;
+using Infrastructure.Context;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Services.OfficeService
+{
+    public class OfficeService : IOfficeService
+    {
+
+        private readonly ApplicationDbContext _dataContext;
+        private readonly IMapper _mapper;
+
+        public OfficeService(ApplicationDbContext dataContext,IMapper mapper)
+        {
+            _dataContext = dataContext;
+            _mapper = mapper;
+        }
+        public async Task<ServiceResponse<List<GetOfficeDTO>>>  AddOffice(AddOfficeDTO office)
+        {
+            Office office1 = new Office
+            {
+                Id = Guid.NewGuid(),
+                Country = office.Country,
+                City = office.City,
+                Address = office.Address,
+                Name = office.Name,
+            };
+
+            var serviceResponse = new ServiceResponse<List<GetOfficeDTO>>();
+            _dataContext.Offices.Add(office1);
+            await _dataContext.SaveChangesAsync();
+            var list = await _dataContext.Offices.ToListAsync();
+            serviceResponse.Data = list.Select(c=>_mapper.Map<GetOfficeDTO>(c)).ToList();
+            return serviceResponse;
+        }
+
+        public async Task<ServiceResponse<List<GetOfficeDTO>>>  GetAllOffices()
+        {
+            var serviceResponse = new ServiceResponse<List<GetOfficeDTO>>();
+            serviceResponse.Data = (await _dataContext.Offices.ToListAsync()).Select(c=>_mapper.Map<GetOfficeDTO>(c)).ToList();
```

Comment:
firstly assing this  _dataContext.Offices.ToListAsync() to another var and then use this var for mapping